extensions/aerosol_optics/mo_aerosol_optics.F90

```diff
@@ -124,51 +124,48 @@ function load_lut(this, band_lims_wvn, &
     ! LUT coefficient dimensions
     !
     npair = size(merra_aero_bin_lims,dim=1)
-    nbin  = size(aero_salt_ext,dim=1)
-    nrh   = size(aero_salt_ext,dim=2)
-    nband = size(aero_salt_ext,dim=3)
+    nval  = size(aero_salt_tbl,dim=1)
+    nrh   = size(aero_salt_tbl,dim=2)
+    nbin  = size(aero_salt_ext,dim=3)
+    nband = size(aero_salt_ext,dim=4)
     !
     ! Error checking
     !
 !    if(nband /= this%get_nband()) &
 !      error_msg = "aerosol_optics%init(): number of bands inconsistent between lookup tables, spectral discretization"
     error_msg = ''
-    if(size(aero_dust_ext, 1) /= nbin) &
-      error_msg = "aerosol_optics%init(): array aero_dust_ext has the wrong number of size bins"
-    if(size(aero_dust_ext, 2) /= nband) &
-      error_msg = "aerosol_optics%init(): array aero_dust_ext has the wrong number of bands"
-    if(size(aero_salt_ext, 3) /= nband) &
-      error_msg = "aerosol_optics%init(): array aero_salt_ext has the wrong number of bands"
-    if(size(aero_sulf_ext, 2) /= nband) &
-      error_msg = "aerosol_optics%init(): array aero_sulf_ext has the wrong number of bands"
-    if(size(aero_bcar_ext, 1) /= nband) &
-      error_msg = "aerosol_optics%init(): array aero_bcar_ext has the wrong number of bands"
-    if(size(aero_ocar_ext, 1) /= nband) &
+    if(size(aero_dust_tbl, 1) /= nval) &
```

Comment:
@mjiacono This seems much more clear to me now, hopefully also to you. 

Can you use `check_extents` from `mo_rte_array_util` please?

Reply:
Done

Reply:
Can you make the same changes at line 273?

Reply:
Done